### PR TITLE
analytics: add Oracle transaction importer

### PR DESCRIPTION
**Pull request by jbowens**
Add new java service for importing Chain Core transactions
to an Oracle database.

Future work:
 * custom reference data fields
 * high-availability (leader election?)
 * true batch inserts of transactions
 * PL/SQL function for archiving old data
 * alerting / health checks?
 * configuring views / materialized views?

### CODE REVIEW

**Comment by jbowens** [comment]
PTAL

This is a first version without any custom reference data fields.

File: analytics/src/main/java/com/chain/analytics/Schema.java [hunk 1]
```diff
@@ -0,0 +1,213 @@
...
+ * constructed, a Schema is immutable.
+ */
+public class Schema {
+  String mName;
+  List<Column> mColumns;
+  List<String> mPrimaryKey;
+  List<List<String>> mUniqueConstraints;
+
+  /**
+   * Column represetns a single column in a table.
```
Comment (boymanjor) [inline analytics/src/main/java/com/chain/analytics/Schema.java]: "represents"

File: analytics/src/main/java/com/chain/analytics/Importer.java [hunk 2]
```diff
@@ -0,0 +1,354 @@
...
+   * and inserting them into the configured Oracle database. This
+   * function blocks indefinitely. Errors are logged to log4j2.
+   *
+   * TODO(jackson): Support propagating persistent errors?
+   */
+  public void process() {
+    ThreadContext.put("feed_id", mFeed.id);
+    ThreadContext.put("feed_filter", mFeed.filter);
+
+    for (; ; ) {
```
Comment (erykwalder) [inline analytics/src/main/java/com/chain/analytics/Importer.java]: `while (true)` might be nice here

File: analytics/src/main/java/com/chain/analytics/Importer.java [hunk 3]
```diff
@@ -0,0 +1,354 @@
...
+            psOut.setBlob(9, asJsonBlob(output.assetTags));
+            psOut.setString(10, "yes".equals(output.assetIsLocal) ? TRUE : FALSE);
+            psOut.setLong(11, output.amount);
+            psOut.setString(12, output.accountId);
+            psOut.setString(13, output.accountAlias);
+            psOut.setBlob(14, asJsonBlob(output.accountTags));
+            psOut.setString(15, output.controlProgram); // clob
+            psOut.setBlob(16, asJsonBlob(output.referenceData));
+            psOut.setString(17, "yes".equals(output.isLocal) ? TRUE : FALSE);
+            psOut.setString(18, FALSE);
```
Comment (erykwalder) [inline analytics/src/main/java/com/chain/analytics/Importer.java]: At what point will we fix this?

**Comment by erykwalder** [comment]
1 nit 1 question

File: analytics/src/main/java/com/chain/analytics/Importer.java (hunk 3 above)
Comment (jbowens) [inline analytics/src/main/java/com/chain/analytics/Importer.java]: Wow, I meant to did it here but it completely slipped my mind. Will do now.

**Comment by jbowens** [comment]
Thanks @erykwalder, fixed to mark outputs as spent

**Comment by erykwalder** [comment]
LGTM

**Pull request merged**